src/Cms/FilePicker.php

```diff
@@ -55,15 +55,14 @@ public function items()
 		// help mitigate some typical query usage issues
 		// by converting site and page objects to proper
 		// pages by returning their children
-		if (is_a($files, 'Kirby\Cms\Site') === true) {
-			$files = $files->files();
-		} elseif (is_a($files, 'Kirby\Cms\Page') === true) {
-			$files = $files->files();
-		} elseif (is_a($files, 'Kirby\Cms\User') === true) {
-			$files = $files->files();
-		} elseif (is_a($files, 'Kirby\Cms\Files') === false) {
-			throw new InvalidArgumentException('Your query must return a set of files');
-		}
+		$files = match ($files::class) {
```

Comment:
What if an object gets passed that extends these classes, e.g. a custom page model? I think the new check will fail in this case.

Reply:
I think you're right! Probably then there is no good way to work with `::class` and `match`

Reply:
From what I've read, this could work

```php
$files = match (true) {
    $files instanceof Site,
    $files instanceof Page,
    $files instanceof User  => $files->files(),
    $files instanceof Files => $files,

    default => throw new InvalidArgumentException('Your query must return a set of files')
};
```

What do you think?

Reply:
https://github.com/getkirby/kirby/pull/4488